### ISSUE TITLE
`PngBuilder::build` segfault on a SVG with embedded PNG image

### ISSUE THREAD

**Issue by akien-mga** [Samsung/thorvg#1278]
This is a follow-up to https://github.com/godotengine/godot/issues/68822, where we use thorvg 0.8.2 in Godot, which fails to load the specific SVG in this zip: 
[testingSVG_fromInkscape_0.svg.zip](https://github.com/Samsung/thorvg/files/10045038/testingSVG_fromInkscape_0.svg.zip)

It fails loading (without crash) at this step in Godot: https://github.com/godotengine/godot/blob/c9c09ad608754a9025ca227358fca5a02553f2cb/modules/svg/image_loader_svg.cpp#L109-L113

Then I tried to see if thorvg `master` (0aec18dc) could process it by using `svg2png`, which triggers a segfault:

```
$ gdb ./build/src/bin/svg2png/svg2png
(gdb) r testingSVG_fromInkscape_0.svg 
Starting program: /home/akien/tmp/thorvg/build/src/bin/svg2png/svg2png testingSVG_fromInkscape_0.svg
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
[New Thread 0x7ffff794f6c0 (LWP 615859)]
[New Thread 0x7ffff714e6c0 (LWP 615860)]
[New Thread 0x7ffff694d6c0 (LWP 615861)]
[New Thread 0x7ffff614c6c0 (LWP 615862)]
[New Thread 0x7ffff594b6c0 (LWP 615863)]
[New Thread 0x7ffff514a6c0 (LWP 615864)]
[New Thread 0x7ffff49496c0 (LWP 615865)]
[New Thread 0x7ffff41486c0 (LWP 615866)]
SVG: Unsupported attributes used [Elements type: Svg][Id : NO_ID][Attribute: stroke-miterlimit][Value: 10]
SVG: Unsupported attributes used [Elements type: Svg][Id : NO_ID][Attribute: clip-rule][Value: evenodd]
SVG: Unsupported attributes used [Elements type: Svg][Id : NO_ID][Attribute: sodipodi:docname][Value: testingVectornatorExport_fromInkscape.svg]
SVG: Unsupported attributes used [Elements type: Svg][Id : NO_ID][Attribute: inkscape:version][Value: 1.2 (dc2aeda, 2022-05-15)]
SVG: Unsupported attributes used [Elements type: Svg][Id : NO_ID][Attribute: inkscape:export-filename][Value: testingSVG_fromInkscape3.svg]
SVG: Unsupported attributes used [Elements type: Svg][Id : NO_ID][Attribute: inkscape:export-xdpi][Value: 96]
SVG: Unsupported attributes used [Elements type: Svg][Id : NO_ID][Attribute: inkscape:export-ydpi][Value: 96]
SVG: Unsupported elements used [Elements: sodipodi:namedview]
SVG: Unsupported attributes used [Elements type: G][Id : Untitled][Attribute: vectornator:layerName][Value: Untitled]
SVG: Unsupported attributes used [Elements type: Image][Id : NO_ID][Attribute: preserveAspectRatio][Value: none]
SVG: Unsupported attributes used [Elements type: Path][Id : NO_ID][Attribute: sodipodi:type][Value: star]
SVG: Unsupported attributes used [Elements type: Path][Id : NO_ID][Attribute: stop-color][Value: #000000]
SVG: Unsupported attributes used [Elements type: Path][Id : path1149][Attribute: inkscape:flatsided][Value: false]
SVG: Unsupported attributes used [Elements type: Path][Id : path1149][Attribute: sodipodi:sides][Value: 5]
SVG: Unsupported attributes used [Elements type: Path][Id : path1149][Attribute: sodipodi:cx][Value: 505.99533]
SVG: Unsupported attributes used [Elements type: Path][Id : path1149][Attribute: sodipodi:cy][Value: 558.7948]
SVG: Unsupported attributes used [Elements type: Path][Id : path1149][Attribute: sodipodi:r1][Value: 411.36679]
SVG: Unsupported attributes used [Elements type: Path][Id : path1149][Attribute: sodipodi:r2][Value: 205.68336]
SVG: Unsupported attributes used [Elements type: Path][Id : path1149][Attribute: sodipodi:arg1][Value: 1.0065281]
SVG: Unsupported attributes used [Elements type: Path][Id : path1149][Attribute: sodipodi:arg2][Value: 1.6348467]
SVG: Unsupported attributes used [Elements type: Path][Id : path1149][Attribute: inkscape:rounded][Value: 0]
SVG: Unsupported attributes used [Elements type: Path][Id : path1149][Attribute: inkscape:randomized][Value: 0]
SVG: Unsupported attributes used [Elements type: Path][Id : path1149][Attribute: inkscape:transform-center-x][Value: -8.1364704]
SVG: Unsupported attributes used [Elements type: Path][Id : path1149][Attribute: inkscape:transform-center-y][Value: -31.46324]

Thread 1 "svg2png" received signal SIGSEGV, Segmentation fault.
0x0000000000402bd9 in PngBuilder::build (this=<synthetic pointer>, buffer=0x7ffeb2ca5010, height=1032197, width=1045713, fileName="/home/akien/tmp/thorvg/testingSVG_fromInkscape_0.png") at ../src/bin/svg2png/svg2png.cpp:52
52                      image[4 * width * y + 4 * x + 0] = (n >> 16) & 0xff;

(gdb) bt
#0  0x0000000000402bd9 in PngBuilder::build (this=<synthetic pointer>, buffer=0x7ffeb2ca5010, height=1032197, width=1045713, fileName="/home/akien/tmp/thorvg/testingSVG_fromInkscape_0.png")
    at ../src/bin/svg2png/svg2png.cpp:52
#1  Renderer::render (this=this@entry=0x7fffffffc5b8, path=path@entry=0x7fffffffc5dc "/home/akien/tmp/thorvg/testingSVG_fromInkscape_0.svg", w=1045713, h=1032197, 
    dst="/home/akien/tmp/thorvg/testingSVG_fromInkscape_0.png", bgColor=<optimized out>) at ../src/bin/svg2png/svg2png.cpp:133
#2  0x0000000000402d68 in App::renderFile (this=this@entry=0x7fffffffc5b8, path=path@entry=0x7fffffffc5dc "/home/akien/tmp/thorvg/testingSVG_fromInkscape_0.svg") at ../src/bin/svg2png/svg2png.cpp:325
#3  0x0000000000403172 in App::setup (this=this@entry=0x7fffffffc5b8, argc=<optimized out>, argv=<optimized out>) at ../src/bin/svg2png/svg2png.cpp:247
#4  0x0000000000402449 in main (argc=<optimized out>, argv=<optimized out>) at ../src/bin/svg2png/svg2png.cpp:396
```

**Comment by mgrudzinska** [comment]
height and width in this svg file are 1032197 and 1045713 respectively. 
the embedded image size is 1024 x 1024
So the icon is quite small. is it intentional? 
That's one of the reasons why you won't see this icon trying to open this svg file using the Svg.cpp example.

The other reason, why the icon isn't visible in general (I tried to use inkscape) is because it's shifted outside the viewbox.
After removing the transformation from the svg file:
-  the result from inkscape:
![ink](https://user-images.githubusercontent.com/67589014/202922495-74a2b280-fe63-426d-b31d-e4954f38dffc.PNG)
 - the result from ThorVG after #1279 is applied
![tvg_icon](https://user-images.githubusercontent.com/67589014/202923293-7f9371a9-6766-4409-8fdb-abe80293b73f.PNG)

**Comment by metanoder** [comment]
@mgrudzinska 

tldr; 
pertaining to icon size intention...
Icon of folder and star is just a quick mockup to test if bug could be repeated using the same workflow.  

to elaborate...
Original file was intended for exported HTML game for Godot 3, and was nested in the Godot 3 game folders.  
When trying to port game to Godot 4, uploading game file into Godot 4 would cause editor to crash.  
After some poking around, it was isolated to the svg.

Workflow to create icon was copying and paste from vanilla MacBook folder icon (command " i ").  Star on top was generated with various vector graphics programs to mimic game icon.

**Comment by metanoder** [comment]
@mgrudzinska 

Ah...  Little late to the party. Please disregard earlier comment.  

Thanks for the fix!!!

**Comment by akien-mga** [comment]
@metanoder I think the point here is that your SVG icon is really a pathological case that wouldn't have any valid use case in a game, because the SVG canvas is HUGE, and the actual icon is located outside the canvas. Here's how it looks like in Inkscape (it can't even zoom out enough to show the whole canvas):

![image](https://user-images.githubusercontent.com/4701338/203527604-083ea83b-8e66-4fe0-ad55-a30cbba16e8c.png)

You can fix it with File > Document Properties > Resize to content:

![image](https://user-images.githubusercontent.com/4701338/203527864-5849d818-8df8-42c4-88bc-2176902fdfd8.png)

---

I confirm that #1279 fixes the crash in `svg2png` with this file.

I'll follow-up on the Godot specific issue in https://github.com/godotengine/godot/issues/68822.
